src/Plugin/Field/FieldWidget/InlineEntityFormMultiple.php

```diff
@@ -557,18 +557,14 @@ public function extractFormValues(FieldItemListInterface $items, array $form, Fo
         });
       }
 
-      // @todo: remove the duplicate entity save.
       foreach ($values as $delta => &$item) {
+        /** @var \Drupal\Core\Entity\EntityInterface $entity */
+        $entity = $item['entity'];
         if (!empty($item['needs_save'])) {
-          // @todo That unset is no longer needed. Figure out whether removing
-          //   it, is the correct fix.
-          // if (!$item['entity']->id()) {
-          // unset($item['entity']->uuid);
-          // }
-          $this->iefController->save($item['entity'], array());
+          $entity->save();
```

Comment:
I was thinking about removing InlineEntityFormHandlerInterface::save() even before, but decided to postpone it. The thing is that EntityFormInterface (which we need to implement in order to go the form handler way) already defines save() and it's signature conflicts with ours. 

I am aware of the fact that commerce product implementation did some stuff here but I suspect that could also be done via one of the standard core ways (hooks, etc)?